fireant/queries/builder.py

```diff
@@ -283,41 +297,104 @@ class DimensionChoicesQueryBuilder(QueryBuilder):
     """
 
     def __init__(self, dataset, dimension):
-        super(DimensionChoicesQueryBuilder, self).__init__(dataset, dataset.hint_table or dataset.table)
+        super(DimensionChoicesQueryBuilder, self).__init__(dataset, dataset.table)
+
         self._dimensions.append(dimension)
 
         # TODO remove after 3.0.0
         display_alias = dimension.alias + '_display'
         if display_alias in dataset.fields:
             self._dimensions.append(dataset.fields[display_alias])
 
+    def _extract_hint_filters(self):
+        """
+        Extracts filters that can be applied when using the hint table.
+
+        :return:
+            A list of filters.
+        """
+        base_table = self.dataset.table
+        hint_table = self.dataset.hint_table
```

Comment:
The hint table will need to come from the dimension rather than the data set.

Reply:
Moved the hint table to the fields / dimensions.